test/Microsoft.AspNetCore.SignalR.Tests/HubEndpointTests.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.IO.Pipelines;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Hosting.Internal;
+using Microsoft.AspNetCore.SignalR;
+using Microsoft.AspNetCore.Sockets;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Xunit;
+
+namespace Microsoft.AspNetCore.SignalR.Tests
+{
+    public class HubEndpointTests
+    {
+        [Fact]
+        public async Task HubsAreDisposed()
+        {
+            var serviceProvider = CreateServiceProvider();
+            var endPoint = serviceProvider.GetService<HubEndPoint<TestHub>>();
+
+            using (var connectionWrapper = new ConnectionWrapper())
+            {
+                var endPointTask = Task.Run(() => endPoint.OnConnectedAsync(connectionWrapper.Connection));
+
+                Thread.Sleep(100);
```

Comment:
Since you have a handle on the Pipeline that was created and passed into OnConnectedAsync, you can get access to the PipelineReaderWriter and await ReadingStarted.